src/service_discovery_setup.js

```diff
@@ -0,0 +1,270 @@
+/*
+ * Things Gateway App.
+ *
+ * Back end main script.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+// External dependencies
+const dnssd = require('dnssd');
+const config = require('config');
+const Settings = require('./models/settings');
+
+// Internal dependencies
+
+const server = new DNSserviceDiscovery();
+
+
+/**
+ * Our DNS Service Discovery object.
+ *
+ * Creates an object which contains the 'profile' used to have a mDNS service
+ * discovery process running. The object will be instantiated with default
+ * values to ensure it will always work no matter DB or Config state.
+ *
+ * It provides methods to interact with the object. This object is created
+ * quite early on since the starting process in app.js and parts of the UI in
+ * settings_controller.js need to use its methods.
+ *
+ *
+ * @param {}
+ */
+
+function DNSserviceDiscovery() {
+  this.serviceState = false;
+  this.port = config.get('ports.https');
+  this.localDomain = config.get(
+    'settings.defaults.domain.localControl.mdnsServiceDomain');
+  this.localName = config.get(
+    'settings.defaults.domain.localControl.mdnsServiceName');
+  this.protocol = config.get(
+    'settings.defaults.domain.localControl.mdnsTxt.protocol');
+  this.power = config.get(
+    'settings.defaults.domain.localControl.mdnsTxt.power');
+  this.description= config.get(
+    'settings.defaults.domain.localControl.mdnsTxt.desc');
+  const txt = {desc: this.description,
+               protocols: this.protocol,
+               power: this.power};
+  const options = {name: this.localName, host: this.localDomain, txt};
+
+  // Initialize our object and make sure it's not started on object creations.
+  this.dnssdHandle = dnssd.Advertisement(dnssd.tcp('http'), this.port, options);
+  this.dnssdHandle.stop();
+}
+
+
+/**
+ * A small helper method.
+ *
+ * Used to update the internal state of the mDNS service discovery.
+ * Then restart it depending on the boolean 'state' parameter.
+ *
+ * @param {boolean}: True starts the broadcast service.
+ *                   False stops the broadcast service
+ */
+
+DNSserviceDiscovery.prototype.startService = function(state) {
+  this.serviceState = !!state;    // Make sure it's a boolean value
+  if (state) {
+    this.dnssdHandle.start();
+  } else {
+    this.dnssdHandle.stop();
+  }
+  console.log(`Service Discover state changed to: ${state}`);
```

Comment:
I wanted to print to the screen the value that is passed into the function, which could help with debugging since it's putting the 'truth' to the screen, not what the function and object has interpreted as the truth.
I've not changed this yet. Can you explain your thoughts on this comment please?

Reply:
The passed in value is kind of irrelevant. What I was seeing in the log was "Service Discover state changed to: undefined" The behavior was correct, but the log was kind of useless.